python/cudf_polars/cudf_polars/dsl/expressions/aggregation.py

```diff
@@ -156,22 +164,44 @@ def agg_request(self) -> plc.aggregation.Aggregation:  # noqa: D102
     def _reduce(
         self, column: Column, *, request: plc.aggregation.Aggregation, stream: Stream
     ) -> Column:
-        if (
-            self.name in {"mean", "median"}
-            and plc.traits.is_fixed_point(column.dtype.plc_type)
-            and self.dtype.plc_type.id() in {plc.TypeId.FLOAT32, plc.TypeId.FLOAT64}
+        is_mean_or_median = self.name in {"mean", "median"}
+        is_quantile = self.name == "quantile"
+
+        out_dtype = self.dtype
+        if plc.traits.is_fixed_point(column.dtype.plc_type) and (
+            is_mean_or_median or is_quantile
         ):
-            column = column.astype(self.dtype, stream=stream)
+            cast_to = (
+                self.dtype
+                if is_mean_or_median
+                and plc.traits.is_floating_point(self.dtype.plc_type)
+                else DataType(pl.Float64())
+            )
+            column = column.astype(cast_to, stream=stream)
+            out_dtype = cast_to
+        if column.size == 0 or column.null_count == column.size:
+            res = None
+            if self.name == "n_unique":
+                res = 0 if column.size == 0 else 1
+            return Column(
+                plc.Column.from_scalar(
+                    plc.Scalar.from_py(res, out_dtype.plc_type, stream=stream),
+                    1,
+                    stream=stream,
+                ),
+                name=column.name,
+                dtype=out_dtype,
+            )
```

Comment:
Did Polars not have special handling for all-null columns before?

Reply:
I reverted these changes in aggregation.py. I xfailed the polars test and will fix in a separate PR.